### PR TITLE
Make test case on range stricter

### PR DESCRIPTION
**Pull request by xiaowuhu**
Signed-off-by: xiaowuhu <xiaowuhu@microsoft.com>

- for the previous test case, if tensor.__index__() return 1, can also pass.
- Also updated user call to convert float and int to Tensor

### CODE REVIEW

**Comment by justinchuby** [comment]
I would also make the PR title more descriptive of the change. I took the liberty to update it.

File: onnxscript/test/loop_test.py [hunk 1]
```diff
@@ -21,6 +22,11 @@ def sumprod(x: FLOAT["N"], N: INT64) -> (FLOAT["N"], FLOAT["N"]):  # noqa: F821
             return sum, prod
 
         self.validate(sumprod)
+        x = np.array([2])
+        M = 3
+        sum, prod = sumprod(x, M)
+        assert(sum==np.array([8]) and prod==np.array([16]))
+
```
[the remark this answers is not in the document]
Comment (gramalingam) [reply onnxscript/test/loop_test.py:29]: I think this is fine for now. I think a couple of extensions could be done, but that may require some more code extension.

(i) Testing eager-mode and execute-model-via-ort simultaneously. (Eg., see: https://github.com/microsoft/onnx-script/blob/main/onnxscript/test/converter_test.py#L544 ).
(ii) Moving script code + associated tests into separate files (Eg., see: https://github.com/microsoft/onnx-script/blob/main/onnxscript/test/functions/attr_test.py ).

I am open to suggestions.

File: onnxscript/test/loop_test.py [hunk 2]
```diff
@@ -4,11 +4,10 @@
 from onnxscript import script
 from onnxscript.onnx_opset import opset15 as op
 from onnxscript.onnx_types import FLOAT, INT64
-from onnxscript.test.common import testutils
 from onnxscript.test.testutils import TestBase
 
 
-class LoopOpTester(testutils.TestBase):
+class LoopOpTester(TestBase):
```
Comment (justinchuby) [inline onnxscript/test/loop_test.py:10]: Prefer `testutils.TestBase`: import only modules https://github.com/microsoft/onnx-script/issues/125

File: onnxscript/test/loop_test.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
+import numpy as np
```
[the remark this answers is not in the document]
Comment (xiaowuhu) [reply onnxscript/test/loop_test.py]: fixed

File: onnxscript/test/loop_test.py [hunk 4]
```diff
@@ -20,8 +21,14 @@
...
 
         self.validate(sumprod)
-
+        x = np.array([2])
+        M = 3
+        sum, prod = sumprod(x, M)
+        self.assertEqual(sum, np.array([8]))
+        self.assertEqual(prod, np.array([16]))
+    
+
```
[the remark this answers is not in the document]
Comment (xiaowuhu) [reply onnxscript/test/loop_test.py]: fixed

[the remark this answers is not in the document]
Comment (xiaowuhu) [reply onnxscript/test/loop_test.py]: fixed

**Pull request merged**